pkg/metrics/collector.go

```diff
@@ -392,25 +407,28 @@ func (c ResourcesMetricsCollector) collectVulnerabilityIdReports(ctx context.Con
 			if c.Config.MetricsVulnerabilityId {
 				vulnLabelValues[0] = r.Namespace
 				vulnLabelValues[1] = r.Name
-				vulnLabelValues[2] = r.Report.Registry.Server
-				vulnLabelValues[3] = r.Report.Artifact.Repository
-				vulnLabelValues[4] = r.Report.Artifact.Tag
-				vulnLabelValues[5] = r.Report.Artifact.Digest
+				vulnLabelValues[2] = r.Labels[trivyoperator.LabelResourceKind]
+				vulnLabelValues[3] = r.Labels[trivyoperator.LabelResourceName]
+				vulnLabelValues[4] = r.Labels[trivyoperator.LabelContainerName]
+				vulnLabelValues[5] = r.Report.Registry.Server
+				vulnLabelValues[6] = r.Report.Artifact.Repository
+				vulnLabelValues[7] = r.Report.Artifact.Tag
+				vulnLabelValues[8] = r.Report.Artifact.Digest
 				for i, label := range c.GetReportResourceLabels() {
-					vulnLabelValues[i+8] = r.Labels[label]
```

Comment:
I believe this was previously set incorrectly since the index for the dynamic labels should have started with 12 (after the `vulnLabelValues` ones), I accounted for this in this PR and also added specific test for this scenario to validate.